packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -607,6 +607,8 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
               );
               currentHighlight = widget.enableSearch ? i : null;
               widget.onSelected?.call(entry.value);
+              _localTextEditingController?.dispose();
+              _localTextEditingController = null;
```

Comment:
Why we need to dispose _localTextEditingController on every click? It will cause recreation of the controller.

Reply:
This getter recreates controller if there is no controller:

```
  TextEditingController get _textEditingController {
    return widget.controller ?? (_localTextEditingController ??= TextEditingController());
  }
```

Reply:
Yes, having getter which is quite troublesome even if we fix this issue if bounds to cause issue if new feature is introduced 

i'll try to remove getter and minimise instantiation limited to `initState` and `didUpdateWidget`